[szelingtan] ip

szelingtan opened the pull request:
# 🤖 🌟 Tringa Bot: Tasks Made Simple 

> - [x] Do you have a **busy** schedule?
> - [x] Do you want a **better** way to manage your schedule?

**Introducing...**  `Tringa`, a bot with 
* _Core Features_: Smart and efficient task management system
* _Acessibility_: Free and open accessibility for everyone

```java
public class Tringa { }
```

**Commands:**
1. _Find_ tasks using keyword search
2. _Add_ tasks with custom categories
3. _View_ task listings and completion status
4. _Mark_ tasks as done
5. _Delete_ unwanted tasks


>Its time to `simplify` your task management journey, with just one [click](https://github.com/szelingtan/ip) !

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

szelingtan committed:
Level 0. Rename, Greet, Exit

szelingtan committed:
Level-1: Echo

szelingtan committed:
Level-2: Add, List

szelingtan committed:
Level-3: Mark as Done

szelingtan committed:
Level-4: ToDo, Event, Deadline

szelingtan committed:
A-TextUiTesting

szelingtan committed:
Level 5. Handle Errors

szelingtan committed:
Level-6: Delete

szelingtan committed:
Refine existing code

szelingtan committed:
Add Storage Class

szelingtan committed:
Storage class: add save functionality

szelingtan committed:
Merge branch 'branch-Level-7'

szelingtan committed:
Add Parser.java

szelingtan committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add Parser.java

szelingtan committed:
Add TaskList.java

szelingtan committed:
Add TaskList.java

szelingtan committed:
Add Ui.java

szelingtan committed:
Add Ui.java

szelingtan committed:
Fix todo task functionality

szelingtan committed:
Fix todo task functionality

szelingtan committed:
Task.java: add comments

szelingtan committed:
Task.java: add comments

szelingtan committed:
Deadline.java: add comments

szelingtan committed:
Deadline.java: add comments

szelingtan committed:
Update oop

szelingtan committed:
Update oop

szelingtan committed:
Merge branch 'branch-more-oop'

* branch-more-oop:
  Update oop
  Deadline.java: add comments
  Task.java: add comments
  Fix todo task functionality
  Add Ui.java
  Add TaskList.java

szelingtan committed:
Add tringaa package

szelingtan committed:
Add tringaa package

szelingtan committed:
Merge branch 'branch-packages'

* branch-packages:
  Add tringaa package

szelingtan committed:
Merge remote-tracking branch 'origin/master'

szelingtan committed:
Add build.gradle

szelingtan committed:
Update build.gradle

szelingtan committed:
Add JUnit tests

szelingtan committed:
Update build.gradle

szelingtan committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Update build.gradle
  Update build.gradle

szelingtan committed:
Add JavaDoc comments

szelingtan committed:
Update coding standard

szelingtan committed:
Update Parser.java

szelingtan committed:
Update TaskList.java

szelingtan committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc comments

szelingtan committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Update coding standard

# Conflicts:
#	src/main/data/tringa.txt

szelingtan committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Update TaskList.java
  Update Parser.java

# Conflicts:
#	src/main/data/tringa.txt

szelingtan committed:
Update code style

szelingtan committed:
Add Checkstyle

szelingtan committed:
Merge branch 'branch-A-Checkstyle'

* branch-A-Checkstyle:
  Add Checkstyle

DESU-CLUB commented on src/main/java/tringaa/Parser.java:
```diff
@@ -0,0 +1,262 @@
+public class Parser {
+    /** Pattern to match the basic command format: command word followed by arguments */
+    private static final Pattern BASIC_COMMAND_FORMAT =
+            Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
```
I love the way you use regex for the commands, this is actually goated

DESU-CLUB commented on src/main/java/tringaa/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+ * Manages a list of tasks and operations on them.
+ */
+public class TaskList {
+    private final List<Task> tasks;
```
This is great naming convention!

DESU-CLUB commented on src/main/java/tringaa/ToDos.java:
```diff
@@ -0,0 +1,31 @@
+ * Represents a To-Do task in the Tringa task management system.
+ * This class extends the basic Task class and adds functionality specific to To-Do items.
+ */
+public class ToDos extends Task {
```
Perhaps a singular ToDo might be better than ToDos? As this is describing the behaviour of a single ToDo task rather than a collection of ToDos.

DESU-CLUB commented on src/main/java/tringaa/Storage.java:
```diff
@@ -0,0 +1,144 @@
+                    throw new TaskStorageException("Invalid deadline format: " + line);
+                }
+                LocalDate date = LocalDate.parse(parts[3], inputFormatter);
+                yield new Deadline(description, date.toString());
```
Love how you use yield instead of the standard break

DESU-CLUB reviewed (commented):
Overall pretty good work! I'm impressed with the way you used regex and the yield keyword: Those are a hallmark of good coding experience! On top of that, there was little to no code smells in the code, congrats!

Jiayouzz for IP!

szelingtan committed:
Create GUI

szelingtan committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Create GUI

szelingtan committed:
MainWindow.java: Initialise greeting

Neilchen863 commented on src/main/java/tringaa/TaskList.java:
```diff
@@ -113,4 +113,29 @@ private void validateIndex(int index) throws TringaException {
         }
     }
 
+    public String findTasks(String keyword) throws TringaException {
```
I understand that your chatbot is named Tringa but I think the exception TringaException is a bit unclear? cuz I don't really understand what kind of exception it represents.

Neilchen863 commented on src/main/java/tringaa/Parser.java:
```diff
@@ -57,6 +59,7 @@ public static String executeCommand(String input, TaskList tasks, Storage storag
             case "deadline" -> prepareDeadline(arguments, tasks, storage);
             case "event" -> prepareEvent(arguments, tasks, storage);
             case "bye" -> "Bye. Hope to see you again soon!";
+            case "find" -> prepareFindTask(arguments, tasks);
```
Good to see you using "case ->", making the code very clear and readable.

Neilchen863 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,57 @@
+// Storage.java
+import java.io.*;
+import java.util.*;
```
I think the usage of * is not recommended here according to the coding standard.

Neilchen863 commented on src/main/java/Event.java:
```diff
@@ -1,27 +1,26 @@
-    private String start;
-    private String end;
+    private final String start;
+    private final String end;
```
Good to see you using a "final" keyword.

Neilchen863 commented on src/main/java/tringaa/TaskList.java:
```diff
@@ -113,4 +113,29 @@ private void validateIndex(int index) throws TringaException {
         }
     }
 
+    public String findTasks(String keyword) throws TringaException {
+        List<Integer> matchingIndices = new ArrayList<>();
+        String search = keyword.toLowerCase().trim();
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            if (task.getDescription().toLowerCase().contains(search)) {
+                matchingIndices.add(i);
+            }
+        }
+        if (matchingIndices.isEmpty()) {
+            return "No matching tasks found in current list.";
+        } else {
+            StringBuilder sb = new StringBuilder();
+            sb.append("Here are the matching tasks in your " + "list\n");
+            int resultNumber = 1;
+            for (int originalIndex : matchingIndices) {
+                Task task = tasks.get(originalIndex);
+                sb.append(String.format("%d.%s\n", resultNumber, task));
+                resultNumber++;
+            }
+            return sb.toString().trim();
+        }
+
+    }
```
I think there could be some more comments explaining the function here.

Neilchen863 reviewed (commented):
Overall pretty good! Very few coding style violations.

szelingtan committed:
Refactor to ToDo.java

szelingtan committed:
Parser.java: add assertions

szelingtan committed:
Improve code quality

szelingtan committed:
Merge pull request #1 from szelingtan/branch-A-Assertions

Add branch-A-Assertions

szelingtan committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Parser.java: add assertions

szelingtan committed:
Add extension

szelingtan committed:
Merge branch 'branch-Extension'

* branch-Extension:
  Add extension

szelingtan committed:
Merge pull request #2 from szelingtan/branch-A-CodeQuality

Improve code quality

szelingtan committed:
Resolve merge conflicts

szelingtan committed:
Parser.java: Add javadoc comments

The prepareFind method has no Javadoc comments.

The way prepareFind method works is unclear.

Give Javadoc comments.

Javadoc creates a more user-friendly experience as it explains how the method works.

szelingtan committed:
Update Parser.java

szelingtan committed:
Parser.java: Add javadoc comments

prepareFind: missing javadoc comment

Unclear on what the prepareFind method does

Add javadoc comments on what the method does and input arguments

This makes it clearer for readers

szelingtan committed:
Update Reminder.java

szelingtan committed:
Storage.java: Add javadoc comment

szelingtan committed:
Adhere to coding style

Code in Parser.java violates coding style

Not good coding practice

Refer to coding guide to change code and adhere to good coding style

To ensure coding style is properly followed

szelingtan committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  Parser.java: Add javadoc comments

szelingtan committed:
Launcher.java: Update javadocs

Javadocs is unclear

Readers may not understand

Add relevant and helpful details to Javadoc

Creates more insightful Javadocs for reader

szelingtan committed:
Update Duke to Tringa

szelingtan committed:
Add files

szelingtan committed:
Improve GUI

szelingtan committed:
Add UnknownCommandException.java

szelingtan committed:
Add InvalidCommandException.java

szelingtan committed:
Refine code quality

szelingtan committed:
Add screenshot

szelingtan committed:
Change screenshot

szelingtan committed:
Reinstate InvalidCommandException.java

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update MainWindow.java

szelingtan committed:
Update Tringa.java

szelingtan committed:
Add packages

szelingtan committed:
Update Parser.java

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update test files

szelingtan committed:
Refine code

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update TaskList.java

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md

szelingtan committed:
Update README.md
